automation/src/test/scala/org/broadinstitute/dsde/test/api/AuthDomainMultiGroupApiSpec.scala

```diff
@@ -0,0 +1,418 @@
+                    AuthDomainMatcher.checkVisibleNotAccessible(projectName, workspaceName)(user.makeAuthToken())
+                  }
+                }
+              }
+            }
+          }
+
+          "when the user is a billing project owner" - {
+            "can be seen but is not accessible" in {
+
+              val user = UserPool.chooseProjectOwner
+              implicit val authToken: AuthToken = authTokenDefault
+
+              withGroup("AuthDomainOne") { groupOne =>
+                withCleanBillingProject(defaultUser, List(user.email)) { projectName =>
+                  withWorkspace(projectName, "GroupsApiSpec_workspace", Set(groupOne)) { workspaceName =>
+
+                    // user can see workspace but user cannot access workspace
+                    AuthDomainMatcher.checkVisibleNotAccessible(projectName, workspaceName)(defaultUser.makeAuthToken())
+
+                  }(user.makeAuthToken())
+                }
+              }(user.makeAuthToken())
+            }
+          }
+        }
+
+        "when not shared with them" - {
+          "cannot be seen and is not accessible" in {
+
+            val user = UserPool.chooseStudent
+            implicit val authToken: AuthToken = authTokenDefault
+
+            withGroup("AuthDomainOne") { groupOne =>
+              withGroup("AuthDomainTwo", List(user.email)) { groupTwo =>
+
+                withCleanBillingProject(defaultUser) { projectName =>
+                  withWorkspace(projectName, "GroupsApiSpec_workspace", Set(groupOne, groupTwo)) { workspaceName =>
+
+                    // user cannot see workspace and user cannot access workspace
+                    AuthDomainMatcher.checkNotVisibleNotAccessible(projectName, workspaceName)(user.makeAuthToken())
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+
+
+      // in all groups
+      "when the user is in all of the groups" - {
+        "when shared with them" - {
+          "can be seen and is accessible" in {
+
+            val user = UserPool.chooseStudent
+            implicit val authToken: AuthToken = authTokenDefault
+
+            withGroup("AuthDomainOne", List(user.email)) { groupOne =>
+              withGroup("AuthDomainTwo", List(user.email)) { groupTwo =>
+
+                withCleanBillingProject(defaultUser) { projectName =>
+                  withWorkspace(projectName, "GroupsApiSpec_workspace", Set(groupOne, groupTwo), List(AclEntry(user.email, WorkspaceAccessLevel.Reader))) { workspaceName =>
+
+                    // user can see workspace and user can access workspace
+                    AuthDomainMatcher.checkVisibleAndAccessible(projectName, workspaceName, List(groupOne, groupTwo))(user.makeAuthToken())
+                  }
+                }
+              }
+            }
+          }
+
+          "and given writer access" - {
+            "the user has correct permissions" in {
```

Comment:
I feel like this test should also check that the workspace is visible and accessible for the writer, not just that the workspace access level is writer

Reply:
👍